app/services/search_service.rb

```diff
@@ -183,4 +183,37 @@ def textSearch_questions(srchString)
     end
     sresult
   end
+
+  #Search nearby nodes with respect to given latitude and longitude
+  def nearbyNodes(srchString)
+    sresult = DocList.new
+    coordinates = srchString.split(",")
+    lat = coordinates[0]
+    lon = coordinates[1]
+
+    nids = NodeTag.joins(:tag)
+               .where('name LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')
+               .collect(&:nid)
+
+    nids = nids || []
+
+    items = Node.includes(:tag)
+                .references(:node, :term_data)
+                .where('node.nid IN (?) AND term_data.name LIKE ?', nids, 'lon:' + lon[0..lon.length - 2] + '%')
+                .limit(200)
+                .order('node.nid DESC')
+items.each do |match|
```

Comment:
And here maybe slightly better indentation plus spaces around the `=` on line 206, 208 would be really great. 

Thanks again for all your hard work! This is going to be great.

Reply:
Correcting..